src/osyris/spatial/coordinate_transforms.py

```diff
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+from . import utils
+
+
+def change_origin(dataset, new_origin):
+    """
+    Translate all positionnal coordinates to new origin
+    """
+    for g in dataset.groups.keys():
+        for element in dataset[g]:
+            unit = dataset[g][element].unit
+            if unit.is_compatible_with("meter") and hasattr(dataset[g][element],
```

Comment:
You could have a vector that would represent the size of a box, but again, probably uncommon.